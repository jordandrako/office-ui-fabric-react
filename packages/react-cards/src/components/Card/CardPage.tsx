import * as React from 'react';
import { ExampleCard, IComponentDemoPageProps, ComponentPage, PageMarkdown, PropertiesTableSet } from '@uifabric/example-app-base';

import { CardBasicExample } from './examples/Card.Basic.Example';
const CardBasicExampleCode = require('!raw-loader!@uifabric/react-cards/src/components/Card/examples/Card.Basic.Example.tsx') as string;

import { CardCompactExample } from './examples/Card.Compact.Example';
const CardCompactExampleCode = require('!raw-loader!@uifabric/react-cards/src/components/Card/examples/Card.Compact.Example.tsx') as string;

export class CardPage extends React.Component<IComponentDemoPageProps, {}> {
  public render(): JSX.Element {
    return (
      <ComponentPage
        title="Card"
        componentName="Card"
        exampleCards={
          <div>
            <ExampleCard title="Basic Card" code={CardBasicExampleCode}>
              <CardBasicExample />
            </ExampleCard>
            <ExampleCard title="Compact Card" code={CardCompactExampleCode}>
              <CardCompactExample />
            </ExampleCard>
          </div>
        }
        propertiesTables={
          <PropertiesTableSet sources={[require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/Card.types.ts')]} />
        }
        overview={
          <PageMarkdown>{require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/docs/CardOverview.md')}</PageMarkdown>
        }
        bestPractices={<div />}
        dos={<PageMarkdown>{require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/docs/CardDos.md')}</PageMarkdown>}
        donts={<PageMarkdown>{require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/docs/CardDonts.md')}</PageMarkdown>}
        isHeaderVisible={this.props.isHeaderVisible}
      />
    );
  }
}
