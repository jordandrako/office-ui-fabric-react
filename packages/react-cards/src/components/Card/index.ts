export * from './CardItem/CardItem';
export * from './Card';
export * from './Card.types';
