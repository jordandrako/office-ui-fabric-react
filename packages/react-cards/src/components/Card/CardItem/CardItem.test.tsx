import * as React from 'react';
import { mount } from 'enzyme';
import { Card } from '../Card';
import * as renderer from 'react-test-renderer';

describe('Card Item', () => {
  it('can handle not having a class', () => {
    const wrapper = mount(
      <Card>
        <Card.Item>
          <div />
        </Card.Item>
      </Card>
    );

    expect(wrapper.find('.test').length).toBe(0);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Card>
          <Card.Item>
            <div />
          </Card.Item>
        </Card>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when having tokens passed to it', () => {
    const tree = renderer
      .create(
        <Card>
          <Card.Item tokens={{ margin: '12px 8px' }}>
            <div />
          </Card.Item>
        </Card>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
