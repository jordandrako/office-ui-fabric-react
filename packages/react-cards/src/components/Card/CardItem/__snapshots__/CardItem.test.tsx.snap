// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Item renders correctly 1`] = `
<div
  className=
      ms-Stack
      ms-Card
      {
        align-items: space-between;
        border-radius: 2px;
        box-shadow: 0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108);
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        height: inherit;
        justify-content: space-between;
        max-width: 286px;
        min-width: 212px;
        padding-bottom: 12px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 12px;
        transition: box-shadow 0.5s ease;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
      }
      & > *:not(:first-child) {
        margin-top: 12px;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 1;
      }
      @media screen and (-ms-high-contrast: active){& {
        box-shadow: 0 1.6px 3.6px 0 Highlight, 0 0.3px 0.9px 0 Highlight;
      }
>
  <div
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 1;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          width: auto;
        }
  >
    <div />
  </div>
</div>
`;

exports[`Card Item renders correctly when having tokens passed to it 1`] = `
<div
  className=
      ms-Stack
      ms-Card
      {
        align-items: space-between;
        border-radius: 2px;
        box-shadow: 0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108);
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        height: inherit;
        justify-content: space-between;
        max-width: 286px;
        min-width: 212px;
        padding-bottom: 12px;
        padding-left: 12px;
        padding-right: 12px;
        padding-top: 12px;
        transition: box-shadow 0.5s ease;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
      }
      & > *:not(:first-child) {
        margin-top: 12px;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 1;
      }
      @media screen and (-ms-high-contrast: active){& {
        box-shadow: 0 1.6px 3.6px 0 Highlight, 0 0.3px 0.9px 0 Highlight;
      }
>
  <div
    className=
        ms-StackItem
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          flex-shrink: 1;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: auto;
          margin-bottom: 12px;
          margin-left: 8px;
          margin-right: 8px;
          margin-top: 12px;
          width: auto;
        }
  >
    <div />
  </div>
</div>
`;
